src/core/options-watcher.cpp

```diff
@@ -16,6 +16,7 @@ options_watcher::options_watcher( std::chrono::milliseconds update_interval )
 
 options_watcher::~options_watcher()
 {
+    _alive = false;
```

Comment:
Don't need to clear `_options` inside a mutex any more; don't need the try-catch, either.

Reply:
Not locking before clear. try-catch needed because clear and join might throw.

Reply:
But you don't need the clear()...

Reply:
And if join might throw(), please catch it inside stop()...

Reply:
I prefer to clear explicitly, but locking is not needed.
Also found out clear has a `noexcept` overload. Catching `join` errors in `stop`.

Reply:
I'm sorry, I don't agree - if you clear, then a lock is necessary.
You don't need to clear.

Reply:
Removed